SP/Gpu.cpp

```diff
@@ -123,7 +123,7 @@ pair<T, T> root1(u32 N, u32 k) {
     assert(!(N&7));
     assert(k <= N/8);
     N /= 2;
-    long double angle = - M_PIl * k / N;
+    ldouble angle = - M_PIl * k / N;
     return {cosl(angle), sinl(angle)};    
```

Comment:
I don't have a good solution here. The MacOS `long double` is equivalent to `double`, and functions like `cosl` expect `long double` i.e. `double`.